core/src/test/java/org/apache/iceberg/TestSchemaUpdate.java

```diff
@@ -1342,18 +1348,18 @@ public void testSetIdentifierFieldsFails() {
             .setIdentifierFields("locations")
             .apply());
 
-    AssertHelpers.assertThrows("add a map key nested field should fail",
+    AssertHelpers.assertThrows("add an optional field should fail",
         IllegalArgumentException.class,
-        "must not be nested in " + SCHEMA.findField("locations"),
+        "not a required field",
         () -> new SchemaUpdate(SCHEMA, SCHEMA_LAST_COLUMN_ID)
-            .setIdentifierFields("locations.key.zip")
+            .setIdentifierFields("data")
             .apply());
 
-    AssertHelpers.assertThrows("add a map value nested field should fail",
+    AssertHelpers.assertThrows("add a map key nested field should fail",
         IllegalArgumentException.class,
         "must not be nested in " + SCHEMA.findField("locations"),
         () -> new SchemaUpdate(SCHEMA, SCHEMA_LAST_COLUMN_ID)
-            .setIdentifierFields("locations.value.lat")
```

Comment:
Was the test case using `locations.value.lat` removed?

Reply:
Yeah because that will hit the double check before hitting the map check. I added another one later with column `new_map.value.val_col`